Assets/Mirror/Tests/SyncDictionaryTest.cs

```diff
@@ -0,0 +1,213 @@
+            toList.OnDeserializeAll(reader);
+        }
+
+        private void SerializeDeltaTo<T>(T fromList, T toList) where T : SyncObject
+        {
+            NetworkWriter writer = new NetworkWriter();
+            fromList.OnSerializeDelta(writer);
+            NetworkReader reader = new NetworkReader(writer.ToArray());
+            toList.OnDeserializeDelta(reader);
+            fromList.Flush();
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+            serverSyncDictionary = new SyncDictionaryIntString();
+            clientSyncDictionary = new SyncDictionaryIntString();
+
+            // add some data to the list
+            serverSyncDictionary.Add(0, "Hello");
+            serverSyncDictionary.Add(1, "World");
+            serverSyncDictionary.Add(2, "!");
+            SerializeAllTo(serverSyncDictionary, clientSyncDictionary);
+        }
+
+        [Test]
+        public void TestInit()
+        {
+            Dictionary<int, string> comparer = new Dictionary<int, string>
+            {
+                [0] = "Hello",
+                [1] = "World",
+                [2] = "!"
+            };
+            Assert.That(clientSyncDictionary[0], Is.EqualTo("Hello"));
+            Assert.That(clientSyncDictionary, Is.EquivalentTo(comparer));
+        }
+
+        [Test]
+        public void TestAdd()
+        {
+            serverSyncDictionary.Add(4, "yay");
+            SerializeDeltaTo(serverSyncDictionary, clientSyncDictionary);
+            Assert.That(clientSyncDictionary.ContainsKey(4), Is.EqualTo(true));
+            Assert.That(clientSyncDictionary[4], Is.EqualTo("yay"));
+        }
+
+        [Test]
+        public void TestClear()
+        {
+            serverSyncDictionary.Clear();
+            SerializeDeltaTo(serverSyncDictionary, clientSyncDictionary);
+            Assert.That(serverSyncDictionary, Is.EquivalentTo(new SyncDictionaryIntString()));
+        }
+
+        [Test]
+        public void TestSet()
+        {
+            serverSyncDictionary[1] = "yay";
+            SerializeDeltaTo(serverSyncDictionary, clientSyncDictionary);
+            Assert.That(clientSyncDictionary[1], Is.EqualTo("yay"));
+        }
+
+        [Test]
+        public void TestConsecutiveSet()
+        {
+            serverSyncDictionary[1] = "yay";
+            serverSyncDictionary[1] = "world";
+            SerializeDeltaTo(serverSyncDictionary, clientSyncDictionary);
+            Assert.That(clientSyncDictionary[1], Is.EqualTo("world"));
+        }
+
+        [Test]
+        public void TestNullSet()
+        {
+            serverSyncDictionary[1] = null;
+            SerializeDeltaTo(serverSyncDictionary, clientSyncDictionary);
+            Assert.That(clientSyncDictionary.ContainsKey(1), Is.EqualTo(true));
+            // Just like SyncList, SyncDictionary should not accept a null value.
+            Assert.That(clientSyncDictionary[1], Is.EqualTo("World"));
```

Comment:
Are you sure about this one?  can you test a regular Dictionary

Reply:
A regular Dictionary accepts a null value (but not key). I figured we would want to follow the same logic as SyncList, but I will change this.